UPGRADING.md

```diff
@@ -116,6 +117,37 @@ method changes in this release to accommodate the different way that
 the new stack tracks data about peers. This change affects users of
 both stacks.
 
+### Using the updated p2p library
+
+The P2P library was reimplemented in this release. The new implementation is
+enabled by default in this version of Tendermint. The legacy implementation is still
+included in this version of Tendermint and can be enabled in the configuration file.
+
+To make use of the legacy P2P implemementation add the following line to your
+server's configuration file:
+
+```toml
+use-legacy = true
+```
+
+#### New p2p queue types
+
+The new p2p implementation enables selection of the queue type to be used for
+passing messages between peers.
+
+The following values may be used when selecting which queue type to use:
+* `fifo`: (default) An unbuffered and lossless queue that passes messages through
+in the order in which they were received.
+* `wdrr`: A queue implementing the Weighted Deficit Round Robin algorithm. A 
+weighted deficit round robin queue is created per peer. Each queue contains a 
+separate 'flow' for each of the channels of communication that exist between any two
+peers. Tendermint maintains a channel per message type between peers. Each WDRR
+queue maintains a shared buffered with a fixed capacity through which messages on different
+flows are passed.
+For more information on WDRR scheduling, see: https://en.wikipedia.org/wiki/Deficit_round_robin
+
+* `priority`: A priority queue of messages.
```

Comment:
I wonder if we should be making this the default before releasing. I believe the intention was to have this as the main approach to queueing messages (cc @alexanderbez)